### PR TITLE
Make Blob/Bucket reload() public and remove the public interface to _properties

### PR DESCRIPTION
**Pull request by dhermes**

### CODE REVIEW

File: gcloud/storage/_helpers.py [hunk 1]
```diff
@@ -82,7 +73,7 @@ def batch(self):
         """
         return _PropertyBatch(self)
 
-    def _reload_properties(self):
+    def reload(self):
```
Comment (tseaver) [inline gcloud/storage/_helpers.py]: Hmm, why are we returning `self` here?

**Comment by tseaver** [comment]
LGTM.  My question about `return self` from mutators shouldn't block the merge, but we should be dropping that pattern pretty much everywhere.

File: gcloud/storage/_helpers.py (hunk 1 above)
Comment (dhermes) [inline gcloud/storage/_helpers.py]: Artifact of JavaScript-y method chaining I would guess. I was just going for the rename here, but removing the return value is easy.

I checked locally and it causes no test failures.

Shall I ditch the return value?

Comment (tseaver) [inline gcloud/storage/_helpers.py]: Or we can create an issue for cleaning those bits out (e.g., `git grep "return self$").

**Comment by tseaver** [comment]
@dhermes needs rebase?

File: gcloud/storage/_helpers.py (hunk 1 above)
Comment (dhermes) [inline gcloud/storage/_helpers.py]: OK cool. I like that.

Comment (dhermes) [inline gcloud/storage/_helpers.py]: Filed #776

**Comment by dhermes** [comment]
@tseaver Rebased and filed the bug about `return self`.

**Pull request merged**